country-dashboard/src/App.tsx

```diff
@@ -1,28 +1,253 @@
-import logo from './logo.svg';
-import './App.css';
-import countries from './data/countries.json';
+import "./App.css";
+
+import Card from "./components/Card";
+import { useEffect, useState } from "react";
 
 export const App = () => {
-  // countries holds a JSON array of the countries and their data which you can use
-  console.log(countries);
+  // Countries state to store the array of countries
+  const [countries, setCountries] = useState<any[]>([]);
+
+  // state to store currency array
+  const [curr, setCurr] = useState<any[]>([]);
+
+  // state to store language array
+  const [lang, setLang] = useState<any[]>([]);
+
+  // state to store region array
+  const [region, setRegion] = useState<any[]>([]);
+
+  // state for the search field to filter
+  const [search, setSearch] = useState("");
+  // state for the currency field to filter
+  const [currency, setCurrency] = useState("Any Currency");
+  // state for the search field to filter
+  const [language, setLanguage] = useState("Any Language");
+  // state for the search field to filter
+  const [reg, setregion] = useState("Any Region");
+
+  // function to get  the countires , currencies , languages , region
+  useEffect(() => {
+    (async () => {
+      const response = await fetch("https://restcountries.com/v3.1/all");
+      const data = await response.json();
+      setCountries(data);
+      setCurr(
+        data
+          ? Array.from(
```

Comment:
While this logic is clever, and I like that (esp. the use of Set), this is very difficult to read. this should have been done in multiple steps using variables . -3